server/src/main/java/org/eclipse/lsp/cobol/core/visitor/VariableDefinitionDelegate.java

```diff
@@ -264,6 +265,45 @@ public void notifySectionChanged() {
     return new ResultWithErrors<>(new ArrayList<>(variables), new ArrayList<>(errors));
   }
 
+  /**
+   * Handle and process REDEFINES clause
+   * @param context is a ParserRuleContext context
+   */
+  public void handleRedefine(DataRedefinesClauseContext context) {
+    String redefinesName = VisitorHelper.getName(context.dataName());
+    Locality locality = positions.get(context.dataName().getStart());
+
+    if (locality != null) {
+      boolean foundRoot = false;
+      Iterator<Variable> iterator = variables.iterator();
+      if (iterator.hasNext()) {
```

Comment:
I would apply early returns in those two cases to decrease nesting level, but it is up to you